### ISSUE TITLE
Quake mode window no longer hides using shortcut key when hide when focus is lost is enabled

### ISSUE THREAD

**Issue by lpinner** [gnunn1/terminix#569]
Quake mode Terminix window no longer hides using shortcut key when hide when focus is lost is enabled.

Ubuntu Gnome 16.04
Terminix 1.3.5 from WebUpd8 PPA

To reproduce:

- Bind a keyboard shortcut to `terminix --quake`
- Enable hide when focus is lost in Terminix preferences
- Press shortcut key to show quake mode Terminix window
- Press shortcut key to hide Terminix

Result:

- Terminix hides then immediately reappears.

Expected behaviour:

- Terminix hides and stays hidden until keyboard shortcut is pressed again.

**Comment by gnunn1** [comment]
Thanks for reporting this. The issue is that the quake window loses focus when you hit the hotkey, so with the "Hide on when focus is lost" option is active it hides the window. Then the hotkey kicks in from the desktop environment and since the window was just hidden it re-shows it again.

I'll need some time to think about how to solve this one, since the hot key is managed by the desktop environment and not terminix it makes things somewhat more complicated. A simple flag won't do the trick.

**Comment by gnunn1** [comment]
I've fixed this (hopefully!) by delaying the hiding of the window on losing focus by 100 ms. I hate using these sort of timing hacks but in this case I can't think of a better solution for the problem.

**Comment by materemias** [comment]
still I have the very same issue with 1.5.0 on Arch

**Comment by gnunn1** [comment]
@materemias I use Arch as well and I can't reproduce on Gnome Shell, what DE are you using?

**Comment by materemias** [comment]
@gnunn1 Gnome Shell. I set up custom key binding and experience just what OP described. Without "Hide on when focus is lost" it is working properly. I have a decent rig if that matters. Is there anything you suggest I shall try?

**Comment by gnunn1** [comment]
Here's the options I have set for Quake mode, can you try the same set if yours differs and see if you get better results.

![screenshot from 2017-02-24 09-56-00](https://cloud.githubusercontent.com/assets/9358238/23307989/93a0ecac-fa77-11e6-91bf-0cc267c5db61.png)

**Comment by materemias** [comment]
interestingly after a reboot it started working normally. or normallishly :) I discovered a new symptom which may be related: if i set the quake console fullscreen, close it and open back again it opens up fullscreen and i cannot turn it back to normal size only after many open-close fullscreen non-full screen cycles.
and now that I have opened and closed it many times the original issue is back. when i close the quake terminal it closes and instantly opens back.

I even tried your suggested settings, made no difference

**Comment by gnunn1** [comment]
Please open a new issue for the fullscreen problem and I'll have a look.

Also, is the original issue related to full-screen, i.e. if you don't use fullscreen does it alway work fine until you fullscreen a few times?

**Comment by materemias** [comment]
I think the 2 issues are connected that's why i haven't opened a new issue yet, and yes, it also happens sporadically after a reboot without switching to fullscreen

**Comment by materemias** [comment]
I found a way to reproduce the fullscreen issue: #801
